target/linux/ath79/dts/qca9563_tplink_tl-wa1201-v2.dts

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca956x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,tl-wa1201-v2", "qca,qca9563";
+	model = "TP-Link TL-WA1201 v2";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		mdio-gpio0 = &mdio2;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "green:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			default-state = "on";
```

Comment:
Why do you need to set default_state "on" here?

Reply:
The linux kernel documentation (`devicetree/bindings/leds/common.yaml`) says that this is the initial state of the led which if not defined is "off". This seemed wrong to me for a power-on led. I looked at the archer c6-v2 which is a similar device that also sets `default-state = "on"`.

Reply:
```
      The initial state of the LED. If the LED is already on or off and the
      default-state property is set the to same value, then no glitch should be
      produced where the LED momentarily turns off (or on). The "keep" setting
      will keep the LED at whatever its current state is, without producing a
      glitch.
```

I understand it as if the LED is off or on when Linux is being booted to avoid glitches, not the behavior. All TP-Link devices I've seen so far have the LED turned off when they boot Linux. So I guess "off" is correct. The behavior of the LEDs at runtime is defined using these rules:
```
		led-boot = &led_power;
		led-failsafe = &led_power;
		led-running = &led_power;
		led-upgrade = &led_power;
```

Reply:
@adschm Can you please explain what this option actually does and why we need to set it at all if we have the led-boot setting?

Reply:
If it really means that the LED will be turned steady off/steady on at the beginning of the boot process, the "on"-entry should be removed in all other DTS-files as OpenWrt uses a blinking pattern for the boot process and as such a steady "on" looks wrong to me...

Reply:
led-boot is user-space, so the default-on from DTS will show much earlier during the boot process. This will be helpful if boot fails early, so the user at least has some indication that the device reached OpenWrt/Linux boot after whatever the initial bootloader does.

Reply:
Ah okay... But then I would say the default "off" is correct as one would assume that the router is running correctly if it is on...

Reply:
No, for the "power" LED I assume the router has "power", and not that it finished boot. So, essentially it should just be always on.

Being strict, having the power LED blink is actually inconsistent, that's what a status LED is for. But typically I recommend to have default-on for both power and status as long as there is only one LED, and default-on for power only if there are power+status available.

Reply:
@adschm Okay, but then we should make it consistent...

Reply:
I try my best to enforce this for new PRs. But altering LED definitions for existing devices is typically not a good idea, since they frequently have assumptions and special cases which determined their LED assignment. From my experience, mass changes to LED definitions frequently create more problems than benefit. But again, this should be discussed in another place.